frontend/server/utils.ts

```diff
@@ -61,3 +61,32 @@ export function loadJSON<T>(filepath?: string, defaultValue?: T): T | undefined
     return defaultValue;
   }
 }
+
+/**
+ * Decode the uri component until the uri cannot be decoded any further.
+ *
+ * @param uri uri string to decode
+ */
+export function consistentDecodeURIComponent(uri: string) {
```

Comment:
I don't think this is the right way to do it.
What if the URI want to have '%' as payload?
and it can also throw exceptions when decoding like the following
```
$ decodeURIComponent('100%25=0')
"100%=0"
$ decodeURIComponent('100%=0')
VM392:1 Uncaught URIError: URI malformed
    at decodeURIComponent (<anonymous>)
    at <anonymous>:1:1
```

I think server must explicitly know how many times it was actually encoded.... actually, client should encode exactly once.

Reply:
Yeah, I agree with u. 

But might need some research. From frontend, shld not be an issue as long as we enforce where uri will be encoded.

But on backend side, my issue is I am not sure about the uri I extract from workflow status with k8s API.

How shld we handle that?

Reply:
That's a good point. I'm not sure about that.

Can we try using the object key directly? (Sounds to me that key shouldn't be encoded, because it was stored as a single field.)
Do you have a case it will be encoded for at least once?

Reply:
Let's scope the issue down to valid S3 URIs that can be generated and used by Argo.

Reply:
let me revert the changes. probably can look at this separately. I will ensure the front-end only encode it once.